src/core_codemods/fix_missing_self_or_cls.py

```diff
@@ -0,0 +1,88 @@
+import libcst as cst
+
+from codemodder.codemods.libcst_transformer import (
+    LibcstResultTransformer,
+    LibcstTransformerPipeline,
+)
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+from core_codemods.api import Metadata, ReviewGuidance
+from core_codemods.api.core_codemod import CoreCodemod
+
+
+class FixMissingSelfOrClsTransformer(LibcstResultTransformer, NameResolutionMixin):
+    change_description = "Add `self` or `cls` parameter to instance or class method."
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.current_class_name = None
+
+    def visit_ClassDef(self, node: cst.ClassDef) -> bool:
+        self.current_class_name = node.name.value
+        return True
+
+    def leave_ClassDef(
+        self, original_node: cst.ClassDef, updated_node: cst.ClassDef
+    ) -> cst.ClassDef:
+        self.current_class_name = None
+        return updated_node
+
+    def leave_FunctionDef(
+        self, original_node: cst.FunctionDef, updated_node: cst.FunctionDef
+    ) -> cst.FunctionDef:
+        if self.current_class_name:
```

Comment:
I think you need to check if this def is directly in the classdef body. The way it is right now, I think it would also change inner functions in nested declarations. For example, in the following:
```python
class A():
  def f(self):
    def g():
      pass
```
I think the codemod would change `g()`.
In any case, it warrants a test case.